addons/mail/models/mail_thread.py

```diff
@@ -3540,13 +3541,17 @@ def _notify_by_email_prepare_rendering_context(self, message, msg_vals=False,
             'record_name': record_name,
             'subtitles': [record_name],
             # user / environment
+            'author_user': author_user,  # User who sends the message
             'company': company,
             'email_add_signature': email_add_signature,
             'lang': lang,
             'signature': signature,
             'website_url': website_url,
             # tools
             'is_html_empty': is_html_empty,
+            # display
```

Comment:
Anyway you can't add parameters in stable. but in master I vote to add those parameters, to make it explicit. It makes the method definition a bit longer, but at least when reading it you know how to control it. Context is very specific and people won't read the method content to discover context keys :) .

However being able to somehow force a value from context could be interesting, e.g. if you go through a mail.compose.message and you want to control rendering without having access to all parameters through fields.

Reply:
I agree with you: Adding "magic" context key like that is not great. I also prefer having a good API.

If I can not add any value here and the mail template won't be updated, I don't see how I can do what's asked of me. **Should I target master then?**

Reply:
Message_post and some methods already support kwargs (because we have too much parameters). But they are not all propagated, they are somehow filtered for each submethod. So if you do "message_post(param=value)", it is ok, until you reach the end of kwargs propagation. Then you can transform parameter into context keys. It is a "middle" solution but that shows in stable that parameters are about to be added and are supported at "main API point". As said they are whitelisted so you have to explicitly allow new parameters in message_post (method cited in other comment _get_notify_valid_parameters).